pkg/controller/node/nodecontroller.go

```diff
@@ -238,6 +289,48 @@ func (nc *NodeController) getCondition(status *api.NodeStatus, conditionType api
 	return nil
 }
 
+func (nc *NodeController) checkPod(obj interface{}) {
+	pod, ok := obj.(*api.Pod)
+	if !ok {
+		return
+	}
+	if pod.DeletionTimestamp == nil {
+		return
+	}
+
+	if len(pod.Spec.NodeName) == 0 {
+		nc.forcefullyDeletePod(pod)
+		return
+	}
+
+	nodeObj, found, err := nc.nodeStore.GetByKey(pod.Spec.NodeName)
+	if err != nil {
+		// what do
+		util.HandleError(err)
+		nc.podController.Requeue(pod)
+		return
+	}
+
+	if !found {
+		nc.forcefullyDeletePod(pod)
```

Comment:
We don't - we opened an issue to add it.

On Tue, Oct 20, 2015 at 5:19 PM, David Oppenheimer <notifications@github.com
> wrote:

> In pkg/controller/node/nodecontroller.go
> <https://github.com/kubernetes/kubernetes/pull/15930#discussion_r42556056>
> :
>
> > +
> > +	if len(pod.Spec.NodeName) == 0 {
> > +		nc.forcefullyDeletePod(pod)
> > +		return
> > +	}
> > +
> > +	nodeObj, found, err := nc.nodeStore.GetByKey(pod.Spec.NodeName)
> > +	if err != nil {
> > +		// what do
> > +		util.HandleError(err)
> > +		nc.podController.Requeue(pod)
> > +		return
> > +	}
> > +
> > +	if !found {
> > +		nc.forcefullyDeletePod(pod)
>
> I could have sworn we already had code somewhere that deleted pods that
> are on nodes that no longer exist, but it wasn't in the GC controller so
> maybe I'm just hallucinating?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/15930/files#r42556056>.
>